sros2/sros2/api/__init__.py

```diff
@@ -291,48 +144,53 @@ def create_permission(keystore_path, identity, policy_file_path):
 
 
 def create_permissions_from_policy_element(keystore_path, identity, policy_element):
-    domain_id = os.getenv(DOMAIN_ID_ENV, '0')
     relative_path = os.path.normpath(identity.lstrip('/'))
-    key_dir = os.path.join(keystore_path, KS_CONTEXT, relative_path)
+    key_dir = os.path.join(_keystore.keystore_context_dir(keystore_path), relative_path)
```

Comment:
Since we're refactoring everything anyway, would it make sense to sub in pathlib for readability?

Reply:
That's the thing though-- we're not actually refactoring anything here. That would require a more in-depth review and this simple re-organization can already be a pretty significant diff to look through. As it stands I can promise that no meaningful changes were made, which makes review a bit faster. Once this re-organization is finished, we can change to using pathlib without breaking API, so it isn't time-sensitive.